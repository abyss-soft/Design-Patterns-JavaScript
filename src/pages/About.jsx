import React from 'react';
import styled from 'styled-components';

const StyledAbout = styled.div`
  color: ${props => props.theme.text};

  a {
    border-bottom: 1px solid ${props => props.theme.CRIMSON};
    color: ${props => props.theme.CRIMSON};
    text-decoration: none;

    &:hover {
      border-bottom: 1px solid transparent;
    }
  }
`;

const Header = styled.h3`
  color: ${props => props.theme.header};
  margin-top: 2rem;
`;

const About = () => (
  <StyledAbout>
    <Header>The Game</Header>
    <p>
      Design Patterns - get familiar with the design patterns implemented in JavaScript, test
      yourself (or someone else) or simply for fun. Enjoy!
    </p>

    <Header>References</Header>
    <p>
      All the code samples are taken from this{' '}
      <cite>
        <a href="//github.com/fbeline/Design-Patterns-JS" target="_blank">
          awesome code compilation
        </a>{' '}
        by Felipe Beline
      </cite>
      .
    </p>
    <p>
      If you want a deep dive into the subject I can't recommend enough{' '}
      <cite>
        <a href="//addyosmani.com/resources/essentialjsdesignpatterns/book/" target="_blank">
          Learning JavaScript Design Patterns
        </a>{' '}
        by Addy Osmani
      </cite>
      .
    </p>
  </StyledAbout>
);

export default About;
