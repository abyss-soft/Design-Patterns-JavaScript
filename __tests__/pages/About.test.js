import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { themeLight } from '../../src/styles/themes/theme.light';
import { themeDark } from '../../src/styles/themes/theme.dark';
import About from '../../src/pages/About';

describe('About page', () => {
  it('renders with a LIGHT theme', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={themeLight}>
          <About />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with a DARK theme', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={themeDark}>
          <About />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
