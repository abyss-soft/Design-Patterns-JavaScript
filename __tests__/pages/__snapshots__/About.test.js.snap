// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page renders with a DARK theme 1`] = `
.c0 {
  color: #C8C8C8;
}

.c0 a {
  border-bottom: 1px solid;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:hover {
  border-bottom: 1px solid transparent;
}

.c1 {
  color: #F5F5F5;
  margin-top: 2rem;
}

<div
  className="c0"
>
  <h3
    className="c1"
  >
    The Game
  </h3>
  <p>
    Design Patterns - get familiar with the design patterns implemented in JavaScript, test yourself (or someone else) or simply for fun. Enjoy!
  </p>
  <h3
    className="c1"
  >
    References
  </h3>
  <p>
    All the code samples are taken from this
     
    <cite>
      <a
        href="//github.com/fbeline/Design-Patterns-JS"
        target="_blank"
      >
        awesome code compilation
      </a>
       
      by Felipe Beline
    </cite>
    .
  </p>
  <p>
    If you want a deep dive into the subject I can't recommend enough
     
    <cite>
      <a
        href="//addyosmani.com/resources/essentialjsdesignpatterns/book/"
        target="_blank"
      >
        Learning JavaScript Design Patterns
      </a>
       
      by Addy Osmani
    </cite>
    .
  </p>
</div>
`;

exports[`About page renders with a LIGHT theme 1`] = `
.c0 {
  color: #6F256F;
}

.c0 a {
  border-bottom: 1px solid;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:hover {
  border-bottom: 1px solid transparent;
}

.c1 {
  color: #6F256F;
  margin-top: 2rem;
}

<div
  className="c0"
>
  <h3
    className="c1"
  >
    The Game
  </h3>
  <p>
    Design Patterns - get familiar with the design patterns implemented in JavaScript, test yourself (or someone else) or simply for fun. Enjoy!
  </p>
  <h3
    className="c1"
  >
    References
  </h3>
  <p>
    All the code samples are taken from this
     
    <cite>
      <a
        href="//github.com/fbeline/Design-Patterns-JS"
        target="_blank"
      >
        awesome code compilation
      </a>
       
      by Felipe Beline
    </cite>
    .
  </p>
  <p>
    If you want a deep dive into the subject I can't recommend enough
     
    <cite>
      <a
        href="//addyosmani.com/resources/essentialjsdesignpatterns/book/"
        target="_blank"
      >
        Learning JavaScript Design Patterns
      </a>
       
      by Addy Osmani
    </cite>
    .
  </p>
</div>
`;
